int16 val
int16 incrementAmount
